### ISSUE TITLE
Framework: Speeding Up The Tests

### ISSUE THREAD

**Issue by gziolo** [Automattic/wp-calypso#3822]
This issues replaces #2108 where we have collected what we found out on this topic.

### Action items
During last Calypso hangout we agreed to the following action Items:
* [x] [look into switching test reporter](https://github.com/Automattic/wp-calypso/pull/4038) (@blowery, @gziolo)
* [x] [look into parallelism](https://github.com/Automattic/wp-calypso/pull/4038) (@blowery, @gziolo)
* [x] @mtias will wrangle developers to help port and modify their tests to single runner
* [x] write P2 to urge more developers to writing single test runners in higher level folders (@blowery)
* [x] ~~update docs with what they’ve learned so far~~ - P2 mentioned in bullet point above covers also that (@blowery)
* [x] @umurkontaci and @gziolo - check old PRs related to single runner

### High level tasks
Other tasks we are working on in parallel:
* [x] [Framework: Enable Parallel Tests on Circle-CI](https://github.com/Automattic/wp-calypso/pull/3793) (@blowery)
* [x] [Framework: Add a client-wide test suite](https://github.com/Automattic/wp-calypso/pull/3797) (@blowery)
* [x] [Add rules for i18n linting](https://github.com/Automattic/eslint-plugin-wpcalypso/pull/3) +  [Replace i18nlint with custom ESLint rules](https://github.com/Automattic/wp-calypso/pull/3849) (@aduth)
* [x] [Move test runner to test folder](https://github.com/Automattic/wp-calypso/pull/3856) (@blowery)
* [x] [Move test helpers into test/test/helpers](https://github.com/Automattic/wp-calypso/pull/3894)(@blowery)
* [x] [Extract `auto` from `lib/react-test-env-setup/index.js` into `test/helpers/use-fake-dom`](https://github.com/Automattic/wp-calypso/pull/3857) ([Related discussion](https://github.com/Automattic/wp-calypso/pull/3797/files#r55208177)) (@gziolo)
* [x] [Consider moving single runner from Makefile to `package.json` and add to Circle CI as `npm test`](https://github.com/Automattic/wp-calypso/pull/4038) (@gziolo)
* [x] [Migrate tests from Makefile to single test runner](https://github.com/Automattic/wp-calypso/issues/3942) - (@aduth, @blowery, @gziolo)
* [x] [Update all docs containing content related to unit tests](https://github.com/Automattic/wp-calypso/pull/4209)
* [x] [Remove no longer needed code (`tests.json` configuration files, shell scripts, CircleCI setup) once all tests are migrated](https://github.com/Automattic/wp-calypso/pull/4596) (@gziolo)
* [x] [Enable parallel execution of single test runner on all CircleCI containers](https://github.com/Automattic/wp-calypso/pull/4185) (@gziolo)
* [x] ~~Make sure `useFilesystemMocks` and `useMockery` work properly together~~ - we should avoid using `useFilesystemMocks` in new tests and we should be good :)
* [x] ~~Consider using tear-down quick mocks between tests ([Related discuusion](https://github.com/Automattic/wp-calypso/pull/3797#discussion_r55210700))~~ We decided to take YAGNI approach, we are going to implement it when there is first use case

**Comment by gziolo** [comment]
>  check old PRs related to single runner

We are heading in the completely different direction, so it doesn't look like we can reuse old migration scripts.

**Comment by gziolo** [comment]
We can consider it done, follow up tasks are listed in #4915.
